### PR TITLE
Remove legacy MBQL filters

### PR DESCRIPTION
**Pull request by ranquild**
Epic https://github.com/metabase/metabase/issues/50336

Historically the legacy `DatePicker` used `Filter` type incorrectly - it actually worked with a raw array and not a class instance. This PR fixes this issue, and finally removes the `Filter` class completely.

### CODE REVIEW

File: frontend/src/metabase/query_builder/components/view/ViewHeader/components/FilterHeaderButton.unit.spec.tsx [hunk 1]
```diff
@@ -22,21 +22,20 @@ const QUERY_WITH_FILTERS = Question.create({
     database: SAMPLE_DB_ID,
     query: {
       "source-table": ORDERS_ID,
+      filter: [
+        "and",
+        ["time-interval", ["field", ORDERS.CREATED_AT, null], -30, "day"],
```
Comment (uladzimirdev) [inline frontend/src/metabase/query_builder/components/view/ViewHeader/components/FilterHeaderButton.unit.spec.tsx:27]: we shouldn't be using refs like this in unit tests, should we?

**Review by uladzimirdev — APPROVED** [review]
minor comment

File: frontend/src/metabase/query_builder/components/view/ViewHeader/components/FilterHeaderButton.unit.spec.tsx (hunk 1 above)
Comment (ranquild) [reply frontend/src/metabase/query_builder/components/view/ViewHeader/components/FilterHeaderButton.unit.spec.tsx:27]: Yes, for sure. This should be migrated as well.

Comment (ranquild) [reply frontend/src/metabase/query_builder/components/view/ViewHeader/components/FilterHeaderButton.unit.spec.tsx:27]: I'll defer this to other PRs

**Pull request merged**